lib/stan.js

```diff
@@ -344,6 +392,81 @@ Stan.prototype.createConnection = function() {
 };
 
 
+/**
+ * Close stan with an error - this should only be with errors
+ * not related to the stan pings. Closes calling error handler
+ * and close handler.
+ * @see Stan#closeDueToPing()
+ * @param event
+ * @param error
+ * @private
+ */
+Stan.prototype.closeWithError = function(event, error) {
+  this.emit(event, error);
+  this.close();
```

Comment:
I have the same issue with the go client (I just realized), but if there is an error during the connect, it may not make sense to try to send the close request. So the internal close function should not try to send the request, only the public Close() function.

Reply:
refactored a bit differently - getting rid of closeDueToPing - and calling directly closeWithError providing the event to notify. The close notification also goes out.